darts/models/arima.py

```diff
@@ -42,16 +43,22 @@ def __init__(self, p: int = 12, d: int = 1, q: int = 0):
     def __str__(self):
         return 'ARIMA({},{},{})'.format(self.p, self.d, self.q)
 
-    def fit(self, series: TimeSeries):
-        super().fit(series)
+    def fit(self, series: TimeSeries, covariates: Optional[TimeSeries] = None):
+        super().fit(series, covariates)
         series = self.training_series
-        m = staARIMA(series.values(),
-                     order=(self.p, self.d, self.q)) if self.d > 0 else staARMA(series.values(), order=(self.p, self.q))
+        covariates = covariates.values() if covariates else None
+
+        if self.d > 0:
+            m = staARIMA(series.values(), exog=covariates, order=(self.p, self.d, self.q))
```

Comment:
were you able to check that statsmodel expects the variable `exog` to be of the same length as the `endog` ?

Reply:
Yes, if `exog` and `exog` are not the same statsmodel will complain. To have a custom error message, I have added our own check in `ExtendedForecastingModel`